python2/koans/about_with_statements.py

```diff
@@ -106,4 +112,10 @@ def count_lines3(self, file_name):
             return len(f.readlines())
 
     def test_open_already_has_its_own_built_in_context_manager(self):
-        self.assertEqual(__, self.count_lines3("example_file.txt"))
+        self.assertEqual(4, self.count_lines3("example_file.txt"))
+## COMMENT
+#
+# For what I understand, the problem was to manually call the opening and closing of a file and that is
+# what is solved with the Context Manager, however I still have some doubts on how you interpret the 
+# file opened, is like a list of strings? because you appear to move between every word separated by an
+# \n
```

Comment:
Context Managers help with the boilerplate code required to open and close files. Note that while opening and closing files is the most common use case, you can do use context managers to do whatever you want. Another typical example is to open and close a DB connection.

How the file is read is a separate problem, not directly related to Context Managers. Most of Python's file operations read lines, where a line is a string of text terminated by `\n`, as you correctly point out. `readlines` in this case loads the entire file into memory (which can be a problem if the file is large), and then creates a `tuple` with an element for each line found in the text. The [file](https://github.com/stevendes/python_koans/blob/master/python2/example_file.txt) this koan is loading just so happens to have one word per line. If you had one sentence per line, it would have returned each sentence as an element in a tuple, in the same order they're found in the text.